test/e2e/network/conntrack.go

```diff
@@ -557,5 +559,56 @@ var _ = common.SIGDescribe("Conntrack", func() {
 		}
 		framework.Logf("boom-server pod logs: %s", logs)
 		framework.Logf("boom-server OK: did not receive any RST packet")
+
+		e2epod.DeletePodOrFail(ctx, fr.ClientSet, fr.Namespace.Name, serverPod.Name)
+	}
+
+	ginkgo.It("proxy implementation should not be vulnerable to the invalid conntrack state bug [Privileged] [ProxyModeIPVS] [ProxyModeNFTables]", CTStateInvalidBugTest)
+
+	ginkgo.It("iptables proxy should not be vulnerable to the invalid conntrack state bug [Privileged] [ProxyModeIPTables]", func(ctx context.Context) {
```

Comment:
@danwinship @aojea I was planning to add `ProxyModeIPVS` and `ProxyModeNFTables` in the skip list of `pull-kubernetes-e2e-kind` and the same for other proxy modes ci jobs to run implementation-specific tests.
I'm not sure if this is the best way to organize the tests, need some guidance here.

Reply:
I don't think using test tags to select proxy-mode-specific tests is a good plan.

If you're manually curl-ing the metrics port anyway, then you could also just curl `127.0.0.1:10249/proxyMode` and check if it's `iptables`, and that combined with whether tcp-be-liberal was set should tell you whether or not to check the metric.

Alternatively, you could rework metrics registration a little and only register this particular metric if it's actually going to be updated (ie, we're using iptables and created a drop rule). Then you can just try to read the metric, and require that either (a) the metric isn't reported, or (b) it gets updated after doing the test. (Maybe it's considered stylistically wrong to conditionally register metrics though? Do any other components do that?)